firefox_media_tests/playback/test_video_playback.py

```diff
@@ -41,38 +44,25 @@ def test_playback_starts(self):
                                         level='WARNING')
 
     def test_video_playback_partial(self):
-        self.run_playback(timeout=120)
+        self.run_playback(duration=60)
 
     def test_video_playback_full(self):
         self.run_playback()
 
-    def run_playback(self, timeout=0):
+    def run_playback(self, interval=1, duration=None,
+                     stall_wait_time=10):
         with self.marionette.using_context('content'):
             for url in self.test_urls:
-                video = VideoPuppeteer(self.marionette, url)
-                # Some vendors don't start the videos at the beginning,
-                # remembering where you were last time you watched.
-                start_time = video.current_time
+                self.logger.info(url)
+                video = VideoPuppeteer(self.marionette, url,
+                                       interval=interval, duration=duration,
+                                       stall_wait_time=stall_wait_time)
                 verbose_until(Wait(video, timeout=30), video,
                               playback_started)
-                duration_timeout = video.duration * 1.3
-                time_delta = 2
-                if timeout <= 0:
-                    duration = duration_timeout
-                else:
-                    # Looser constraint for partial playback
-                    time_delta = 5
-                    duration = min(timeout, duration_timeout)
 
                 try:
-                    verbose_until(Wait(video, interval=1, timeout=duration),
+                    verbose_until(Wait(video, interval=interval,
+                                       timeout=video.duration+stall_wait_time),
```

Comment:
Thinking out loud: what do you think of giving a looser timeout here? Like `video.duration * 1.3 + stall_wait_time`? There's no harm in waiting a bit longer in case the video completes after all. 

This comes from me seeing a lot of intermittent failures on pf-jenkins that seem to be due only test being too strict for our builders' limited resources. We need to keep such failures to a minimum.